org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/commons/TextDocument.java

```diff
@@ -52,6 +54,20 @@ public void setIncremental(boolean incremental) {
 		}
 	}
 
+	/**
+	 * @param doSymbols the doSymbols to set
+	 */
+	public void setDoSymbols(Boolean doSymbols) {

```

Comment:
Son't cache it the result of excluded doSymbols. We will do that if we see performance problem but for xml file association the match(uri) is done every time and it works great